src/garage/experiment/local_tf_runner.py

```diff
@@ -78,10 +80,18 @@ def __init__(self, sess=None, max_cpus=1):
     def __enter__(self):
         """Set self.sess as the default session.
 
+        Also, snapshotter is created if SnapshotConfig is passed.
+
         Returns:
             This local runner.
 
         """
+
+        if isinstance(self.snapshot_config, SnapshotConfig):
+            self.snapshotter = Snapshotter(self.snapshot_config.snapshot_dir,
```

Comment:
Instead of passing every field of `snapshot_config` into the `Snapshotter`, why not just make `snapshot_config` a constructor function for `Snapshotter`? e.g. this line would become:
```python
self.snapshotter = self.snapshot_config()
```

Reply:
Calling a `config` to create a `snapshotter` is a kind of surprising behaviour to me. Also, I don't quite get your point of:

> users can pass in a new type if they would like

I think our concern here is to know whether user want to snapshot or not. What type would they pass into `LocalRunner` potentially except `SnapshotConfig`?

Reply:
I just checked the doc that both `SimpleNamespace` and `namedtuple` don't have the `__call__` attribute.

Reply:
What I was trying to get at is that users might have a different snapshotter class that they want to use instead of our snapshotter.

Since "calling" classes in Python constructs them, a simple way of making it possible to pass in a new snapshotter type is to make the way that you use the config class be to call it. Then, the user could just pass their own class in place of the config, and everything would just work.

Notably, you can still define `__call__` for a custom named tuple (for example, like this):
```python
import typing

class SnapshotterConfig(typing.NamedTuple):
    snapshot_dir: str
    snapshot_mode: str
    snapshot_gap: int

    def __call__(self):
        return Snapshotter(**self._asdict())
```

Having said that, custom Snapshotters seem like a niche concern, so we can drop it.